app/common/data/interfaces/collections.py

```diff
@@ -33,22 +33,28 @@ def create_collection_schema(*, name: str, user: User, grant: Grant, version: in
     return schema
 
 
-def get_collection_schema(schema_id: UUID, version: int | None = None) -> CollectionSchema:
+def get_collection_schema(
+    schema_id: UUID, version: int | None = None, with_full_schema: bool = False
+) -> CollectionSchema:
     """Get a collection schema by ID and optionally version.
 
     If you do not pass a version, it will retrieve the latest version (ie highest version number).
 
     Note: We may wish to change this behaviour to the latest 'published' version in the future, or some other logic.
     """
+    options = []
+    if with_full_schema:
+        options.append(selectinload(CollectionSchema.sections).selectinload(Section.forms).selectinload(Form.questions))
```

Comment:
Going to add an integration test to make sure these options are appropriately applied and this is eagerly loaded as expected